executor/sample_test.go

```diff
@@ -18,67 +18,62 @@ import (
 	"flag"
 	"fmt"
 	"sync/atomic"
+	"testing"
 
-	. "github.com/pingcap/check"
 	"github.com/pingcap/tidb/ddl"
-	"github.com/pingcap/tidb/domain"
 	"github.com/pingcap/tidb/errno"
 	"github.com/pingcap/tidb/kv"
 	"github.com/pingcap/tidb/session"
 	"github.com/pingcap/tidb/sessionctx/variable"
 	"github.com/pingcap/tidb/store/mockstore"
-	"github.com/pingcap/tidb/util/testkit"
+	"github.com/pingcap/tidb/testkit"
+	"github.com/stretchr/testify/require"
 	"github.com/tikv/client-go/v2/testutils"
 )
 
-var _ = SerialSuites(&testTableSampleSuite{})
-
-type testTableSampleSuite struct {
-	cluster testutils.Cluster
-	store   kv.Storage
-	domain  *domain.Domain
+func createSampleTestkit(t *testing.T, store kv.Storage) *testkit.TestKit {
+	atomic.StoreUint32(&ddl.EnableSplitTableRegion, 1)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("drop database if exists test_table_sample;")
+	tk.MustExec("create database test_table_sample;")
+	tk.MustExec("use test_table_sample;")
+	tk.MustExec("set @@global.tidb_scatter_region=1;")
+	return tk
 }
 
-func (s *testTableSampleSuite) SetUpSuite(c *C) {
+func createSampleStore(t *testing.T) (store kv.Storage, tk *testkit.TestKit, clean func()) {
+	var err error
 	flag.Lookup("mockTikv")
 	useMockTikv := *mockTikv
+
 	if useMockTikv {
-		store, err := mockstore.NewMockStore(
+		store, err = mockstore.NewMockStore(
 			mockstore.WithClusterInspector(func(c testutils.Cluster) {
 				mockstore.BootstrapWithSingleStore(c)
-				s.cluster = c
 			}),
 		)
-		c.Assert(err, IsNil)
-		s.store = store
+		require.NoError(t, err)
 		session.SetSchemaLease(0)
 		session.DisableStats4Test()
 	}
-	d, err := session.BootstrapSession(s.store)
-	c.Assert(err, IsNil)
-	d.SetStatsUpdating(true)
-	s.domain = d
-}
-
-func (s *testTableSampleSuite) TearDownSuite(c *C) {
-	s.domain.Close()
-	c.Assert(s.store.Close(), IsNil)
-}
-
-func (s *testTableSampleSuite) initSampleTest(c *C) *testkit.TestKit {
-	atomic.StoreUint32(&ddl.EnableSplitTableRegion, 1)
-	tk := testkit.NewTestKit(c, s.store)
-	tk.MustExec("drop database if exists test_table_sample;")
-	tk.MustExec("create database test_table_sample;")
-	tk.MustExec("use test_table_sample;")
-	tk.MustExec("set @@global.tidb_scatter_region=1;")
-	return tk
+	dom, err := session.BootstrapSession(store)
+	require.NoError(t, err)
+	dom.SetStatsUpdating(true)
+
+	tk = createSampleTestkit(t, store)
+	clean = func() {
+		dom.Close()
+		err := store.Close()
+		require.NoError(t, err)
+	}
+	return
 }
 
-func (s *testTableSampleSuite) TestTableSampleBasic(c *C) {
-	tk := s.initSampleTest(c)
+func TestTableSampleBasic(t *testing.T) {
+	_, tk, clean := createSampleStore(t)
```

Comment:
should use 

```
store, clean := testkit.CreateMockStore(t)
defer clean()
tk := testkit.NewTestKit(t, store)
```

Reply:
thank your for review.
i see the original init logic in `SetUpSuite` almost the same as `testkit.CreateMockStore(t)` except the latter one missing flag `mockTikv`, which means after migrate the flag is not used and alwasy use mockStore. 
Is this acceptable? I am not sure this part.

Reply:
Yes, CreateMockStore can support the ```mockTikv```.  You can replace it.

Reply:
fxied.
use `testkit.CreateMockStore` but keep `createSampleStore` to remove duplicated code like create and use database.